tests/integration/helpers.py

```diff
@@ -28,6 +28,35 @@
 METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
 DATABASE_APP_NAME = METADATA["name"]
 
+charm = None
+
+
+async def build_and_deploy(
+    ops_test: OpsTest, num_units: int, app_name: str = DATABASE_APP_NAME, status: str = "active"
+) -> None:
+    """Builds the charm and deploys a specified number of units."""
+    global charm
+    if not charm:
```

Comment:
I can't see anywhere else this global is used

Reply:
It's been used to share the already built charm between multiple tests on `tests/integration/test_charm.py` and also on `tests/integration/test_db.py`.

Reply:
TIL. It looks it is a common story for us:

canonical/mysql-k8s-operator/tests/integration/high_availability/high_availability_helpers.py
147:    global mysql_charm
193:    global application_charm